scripts/scripts_utils.py

```diff
@@ -85,69 +106,93 @@ def _download(url: str, local_path: Path) -> Optional[int]:
     return None
 
 
-def get_release(release: Release) -> str:
-    """Install a file to carbon-lang's cache.
-
-    release: The release to cache.
-    """
+def _get_cached_binary(name: str, url: str, want_hash: str) -> str:
     cache_dir = Path.home().joinpath(".cache", "carbon-lang-scripts")
     cache_dir.mkdir(parents=True, exist_ok=True)
 
-    # Translate platform information into Bazel's release form.
-    machine = platform.machine()
-    if machine == "x86_64":
-        machine = "amd64"
-    version = f"{platform.system().lower()}-{machine}"
-
-    # Get ready to add .exe for Windows.
-    ext = ""
-    if platform.system() == "Windows":
-        ext = ".exe"
-
-    # Ensure the platform is supported, and grab its hash.
-    if version not in _VERSION_SHAS[release.value]:
-        # If this because a platform support issue, we may need to print errors.
-        exit(f"No {release.value} release available for platform: {version}")
-    want_hash = _VERSION_SHAS[release.value][version]
-
     # Hold a lock while checksumming and downloading the path. Otherwise,
     # parallel runs by pre-commit may conflict with one another with
     # simultaneous downloads.
-    with open(cache_dir.joinpath(f"{release.value}.lock"), "w") as lock_file:
+    with open(cache_dir.joinpath(f"{name}.lock"), "w") as lock_file:
         fcntl.lockf(lock_file.fileno(), fcntl.LOCK_EX)
 
         # Check if there's a cached file that can be used.
-        local_path = cache_dir.joinpath(f"{release.value}{ext}")
+        local_path = cache_dir.joinpath(name)
         if local_path.is_file() and want_hash == _get_hash(local_path):
             return str(local_path)
 
         # Download the file.
-        url = f"{_URL}/{release.value}-{version}{ext}"
         retries = 5
         while True:
             err = _download(url, local_path)
             if err is None:
                 break
             retries -= 1
             if retries == 0:
-                exit(
-                    f"Failed to download {release.value}-{version}: HTTP {err}."
-                )
+                exit(f"Failed to download {url}: HTTP {err}.")
             time.sleep(1)
         local_path.chmod(0o755)
 
         # Verify the downloaded hash.
         found_hash = _get_hash(local_path)
         if want_hash != found_hash:
             exit(
-                f"Downloaded {release.value}-{version} but found sha256 "
+                f"Downloaded {url} but found sha256 "
                 f"{found_hash} ({local_path.stat().st_size} bytes), wanted "
                 f"{want_hash}"
             )
 
     return str(local_path)
 
 
+def _get_machine() -> str:
+    machine = platform.machine()
+    if machine == "x86_64":
+        machine = "amd64"
+    return machine
+
+
+def _get_platform_ext() -> str:
+    return ".exe" if platform.system() == "Windows" else ""
```

Comment:
I'd really prefer to avoid Python's ternaries; while I understand the desire for brevity, the order of expressions really feels difficult to parse.